model_analyzer/config/generate/optuna_run_config_generator.py

```diff
@@ -0,0 +1,302 @@
+            self._create_trial_objectives(trial)
+            run_config = self._create_objective_based_run_config()
+            yield run_config
+            score = self._calculate_score()
+            self._study.tell(trial, score)
+
+    def _create_trial_objectives(self, trial) -> None:
+        # TODO: TMA-1925: Use SearchParameters here
+        self._instance_count = trial.suggest_int("instance_count", 1, 8)
+        self._batch_size = int(2 ** trial.suggest_int("batch_size", 1, 10))
+
+        # TODO: TMA-1884: Need an option to choose btw. concurrency formula and optuna searching
+        self._concurrency = 2 * self._instance_count * self._batch_size
+        if self._concurrency > 1024:
+            self._concurrency = 1024
+
+    def _create_objective_based_run_config(self) -> RunConfig:
+        param_combo = self._create_parameter_combo()
+
+        # TODO: TMA-1927: Add support for multi-model
+        run_config = RunConfig(self._triton_env)
+
+        model_config_variant = BaseModelConfigGenerator.make_model_config_variant(
+            param_combo=param_combo,
+            model=self._models[0],
+            model_variant_name_manager=self._model_variant_name_manager,
+            c_api_mode=self._c_api_mode,
+        )
+
+        # TODO: TMA-1927: Add support for multi-model
+        model_run_config = self._create_model_run_config(
+            model=self._models[0],
+            model_config_variant=model_config_variant,
+        )
+
+        run_config.add_model_run_config(model_run_config=model_run_config)
+
+        return run_config
+
+    def _create_parameter_combo(self) -> Dict[str, Any]:
+        # TODO: TMA-1925: Use SearchParameters here
+        param_combo: Dict["str", Any] = {}
+        param_combo["dynamic_batching"] = {}
+
+        # TODO: TMA-1927: Add support for multi-model
+        kind = "KIND_CPU" if self._models[0].cpu_only() else "KIND_GPU"
+        param_combo["instance_group"] = [
+            {
+                "count": self._instance_count,
+                "kind": kind,
+            }
+        ]
+
+        param_combo["max_batch_size"] = self._batch_size
+
+        return param_combo
+
+    def _calculate_score(self) -> float:
+        if self._last_measurement:
+            score = self._default_measurement.compare_measurements(  # type: ignore
+                self._last_measurement
+            )
+        else:
+            # TODO: TMA-1927: Figure out the correct value for this (and make it a constant)
+            score = -1
+
+        return score
+
+    def _create_default_run_config(self) -> RunConfig:
+        default_run_config = RunConfig(self._triton_env)
+        # TODO: TMA-1927: Add support for multi-model
+        default_model_run_config = self._create_default_model_run_config(
+            self._models[0]
+        )
+        default_run_config.add_model_run_config(default_model_run_config)
+
+        return default_run_config
+
+    def _create_default_model_run_config(
```

Comment:
Is this any different code from other RCGs? If not, we should not be cutting and pasting.

Reply:
Agreed. I've filed TMA-1930 for this work.